uxarray/grid/utils.py

```diff
@@ -121,3 +123,109 @@ def cross_fma(v1, v2):
     y = _fmms(v1[2], v2[0], v1[0], v2[2])
     z = _fmms(v1[0], v2[1], v1[1], v2[0])
     return np.array([x, y, z])
+
+
+def _inv_jacobian(x0, x1, y0, y1, z0, z1, x_i_old, y_i_old):
+    """Calculate the inverse Jacobian matrix for a given set of parameters.
+
+    Parameters
+    ----------
+    x0 : float
+        Description of x0.
+    x1 : float
+        Description of x1.
+    y0 : float
+        Description of y0.
+    y1 : float
+        Description of y1.
+    z0 : float
+        Description of z0.
+    z1 : float
+        Description of z1.
+    x_i_old : float
+        Description of x_i_old.
+    y_i_old : float
+        Description of y_i_old.
+
+    Returns
+    -------
+    numpy.ndarray or None
+        The inverse Jacobian matrix if it is non-singular, or None if a singular matrix is encountered.
+
+    Notes
+    -----
+    This function calculates the inverse Jacobian matrix based on the provided parameters. If the Jacobian matrix
+    is singular, a warning is printed, and None is returned.
+    """
+
+    # d_dx = (x0 * x_i_old - x1 * x_i_old * z0 + y0 * y_i_old * z1 - y1 * y_i_old * z0 - y1 * y_i_old * z0)
+    # d_dy = 2 * (x0 * x_i_old * z1 - x1 * x_i_old * z0 + y0 * y_i_old * z1 - y1 * y_i_old * z0)
+    #
+    # # row 1
+    # J[0, 0] = y_i_old / d_dx
+    # J[0, 1] = (x0 * z1 - z0 * x1) / d_dy
+    # # row 2
+    # J[1, 0] = x_i_old / d_dx
+    # J[1, 1] = (y0 * z1 - z0 * y1) / d_dy
+
+    # The Jacobian Matrix
+    jacobian = [[_fmms(y0, z1, z0, y1),
+                 _fmms(x0, z1, z0, x1)], [2 * x_i_old, 2 * y_i_old]]
+    try:
+        inverse_jacobian = np.linalg.inv(jacobian)
+    except np.linalg.LinAlgError:
+        raise warnings("Warning: Singular Jacobian matrix encountered.")
+        return None
+
+    return inverse_jacobian
+
+
+def _newton_raphson_solver_for_gca_constLat(init_cart,
+                                            gca_cart,
+                                            max_iter=1000,
+                                            verbose=False):
+    """Solve for the intersection point between a great circle arc and a
+    constant latitude.
+
+    Args:
+        init_cart (np.ndarray): Initial guess for the intersection point.
+        w0_cart (np.ndarray): First vector defining the great circle arc.
+        w1_cart (np.ndarray): Second vector defining the great circle arc.
+        max_iter (int, optional): Maximum number of iterations. Defaults to 1000.
+        verbose (bool, optional): Whether to print verbose output. Defaults to False.
+
+    Returns:
+        np.ndarray or None: The intersection point or None if the solver fails to converge.
+    """
+    tolerance = ERROR_TOLERANCE
```

Comment:
It might be better to introduce another tolerance variable for such solver methods.

Reply:
The tolerance must be cohere globally here for better result guarantee.

Instead of a solver method, it's used to correct out the numerical result to make them coheres with our global accuracy